dom/node_test.go

```diff
@@ -239,3 +241,39 @@ func TestCreateElementNS(t *testing.T) {
 
 	t.Logf("%s", doc.Dump(false))
 }
+
+func TestGH102(t *testing.T) {
+	t.Run("child ns after parent ns", func(t *testing.T) {
+		root := CreateDocument()
+		doc, _ := root.CreateElement("Document")
+		_ = doc.SetNamespace("https://example.com", "ex")
+		test, _ := root.CreateElement("Test")
+		_ = doc.AddChild(test)
+		_ = root.SetDocumentElement(doc)
+		_ = test.SetNamespace("https://example.com", "ex")
+		test.SetNodeValue("test")
+
+		// root.Dump(true) should only contain one namespace declaration
+		// for the 'ex' prefix
+		dump := root.Dump(true)
+		count := strings.Count(dump, `xmlns:ex="https://example.com"`)
+		t.Logf("%s", dump)
+		require.Equal(t, 1, count, "expected only one namespace declaration for 'ex' prefix")
+	})
+
+	t.Run("parent ns after child ns", func(t *testing.T) {
+		root := CreateDocument()
+		doc, _ := root.CreateElement("Document")
+		//_ = doc.SetNamespace("https://example.com", "ex") // delete
+		_ = root.SetDocumentElement(doc)
+		test, _ := root.CreateElement("Test")
+		_ = test.SetNamespace("https://example.com", "ex")
+		_ = doc.SetNamespace("https://example.com", "ex") // move to here
+		_ = doc.AddChild(test)
+		test.SetNodeValue("test")
+		dump := root.Dump(true)
+		count := strings.Count(dump, `xmlns:ex="https://example.com"`)
+		t.Logf("%s", dump)
+		require.Equal(t, 2, count, "expected two namespace declaration for 'ex' prefix")
+	})
```

Comment:
Add test to check namespace prefix
```suggestion
	// root.Dump(true) should contains namespace prefix for Document
		count = strings.Count(dump, "ex:Document")
		require.Equal(t, 2, count, "expected Document to have 'ex' namespace prefix")

		// root.Dump(true) should contains namespace prefix for Document
		count = strings.Count(dump, "ex:Test")
		require.Equal(t, 2, count, "expected Document to have 'ex' namespace prefix")
	})
```